packages/search-table-pagination/src/main.vue

```diff
@@ -366,7 +368,19 @@
     watch: {
       data: function(value) {
         this.loadLocalData(value)
-      }
-    }
+      },
+
+      currentPage: function(val) {
+        if (val) {
+          this.pagination.pageIndex = val
```

Comment:
对对象重新赋值的话，要先取原值，修改属性之后，再把修改后的对象重新赋值。

这里的代码修改一下吧，参考如下：
```suggestion
        
const { pagination} = this;
pagination.pageIndex = Val;

this.pagination = pagination;
```

PageSize属性的修改也同上。